### PR TITLE
TOOL-76 Dep update and minor fixes.

### PR DESCRIPTION
**Pull request by richard-bogdan-bitrise**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#   name = "github.com/x/y"
+#   version = "2.4.0"
+#
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
```
Comment (lszucs) [inline Gopkg.toml]: Generated example code should be removed.

**Review by lszucs — CHANGES_REQUESTED** [review]
Minor change request. Nice job cleaning the cruft.

File: main.go [hunk 2]
```diff
@@ -397,3 +403,13 @@ func main() {
 	}
 	// ---
 }
+
+// getStartTime gets the start time of the build.
+func getStartTime() (time.Time, error) {
+	startTimeString := os.Getenv("BITRISE_BUILD_TRIGGER_TIMESTAMP")
```
Comment (godrei) [inline main.go]: why did do you this change? what if there are multiple xamarin-archive steps in the workflow. If the last one runs, but generates 0 artifacts, the step will find the previous artifacts.

**Review by godrei — CHANGES_REQUESTED** [review]
I left 1 comment pls check it.
In general, i do not see how would it fix the failed artifact export, could you pls explain it?

File: main.go (hunk 2 above)
Comment (richard-bogdan-bitrise) [reply main.go]: I haven't thought about this case, but the reason is the following: if there are multiple steps in a workflow, one generates for iOS, one for macOS, the app folder won't be found for the macOS, as the iOS build already generated it (and the mod time will be older). This is why I updated it to the start of the build, but your case could be in issue as well, so have to come up with different solution.

**Comment by richard-bogdan-bitrise** [comment]
The fix can be found in go-xamarin, the parsing itself was updated.

**Comment by richard-bogdan-bitrise** [comment]
@lszucs the CI has been updated, @godrei removed debug related code and fixed the CI, pls verify this and if everything is ok, it can be merged and released.

**Review by lszucs — APPROVED** [review]

**Comment by lszucs** [comment]
@trapacska please merge

**Review by trapacska — APPROVED** [review]

**Pull request merged**